lib/Basics/csv.cpp

```diff
@@ -399,7 +399,7 @@ int TRI_ParseCsvString(TRI_csv_parser_t* parser, char const* line, size_t length
           else {
             parser->_written = qtr;
             parser->_current = ptr;
-            return true;
+            return TRI_ERROR_NO_ERROR;
```

Comment:
I know the previous API returned an integer, which was implicitly converted to an int. The previous implementation returned both true and false.
Now we are returning an ErrorCode, and always return TRI_ERROR_NO_ERROR. I am wondering if that will cause a behavior change for the caller somehow.

Reply:
The error code is ignored everywhere, I shortly thought about changing the return code to `void`. :speak_no_evil: 

I tried to understand the code and return a sane value anyway, but its very possible I got it wrong. My thoughts were the following:

There are three places changed here. The first two are states `TRI_CSV_PARSER_BOL` and `TRI_CSV_PARSER_BOL2`, in both cases meeting the end of input. Both previously returned `false`. I gathered that meeting EOF when at the beginning of a line would be OK. The third place is here, where more input is needed. As this seems to be a reentrant function, and the other places marked with `// need more input` also returned `TRI_ERROR_NO_ERROR`, I thought this was also the right return value here.

Reply:
It is definitely the right thing to return if the return value is ignored anyway as you said :sweat_smile: 
If the return value is really ignored everywhere, I would favor declaring the return value of this function void, but we can do this in a separate PR and there is no need to delay this one.